lib/sycamore/sycamore/materialize.py

```diff
@@ -206,12 +218,14 @@ def _path_to_id(p: Path) -> Optional[str]:
             return None
         return str(p.stem[4:])
 
-    def filter(self, p: str) -> bool:
+    def filter(self, p: str, read: Optional[bool] = False) -> bool:
```

Comment:
Optional[bool] = False doesn't make sense, I'd just do read: bool = False

I'd also make it read_binary since both cases are reads.